src/psadropdown/PSADropdown.jsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  FormControl, InputLabel, MenuItem, Select,
+} from '@mui/material';
+
+export const PSADropdown = ({
+  id,
+  labelId,
+  label,
+  value,
+  onChange,
+  variant,
+  items = [],
+  style = {},
+  formSx = {},
+  inputSx = {},
+  selectSx = {},
+  MenuProps = {},
+  error = false,
+}) => (
+  <FormControl sx={formSx} variant={variant}>
+    <InputLabel
+      id={id}
+      sx={inputSx}
+    >
+      {label}
+    </InputLabel>
+    <Select
+      labelId={labelId}
+      id={id}
+      label={label}
+      value={value}
+      variant={variant}
+      onChange={onChange}
+      MenuProps={MenuProps}
+      style={style}
+      sx={selectSx}
+      error={error}
+    >
+      {items.map((item, index) => (
+        <MenuItem key={index} value={item.value}>
+          {item.label}
```

Comment:
I'm not sure if this is the best way, I just searched the usage in selector and there are some MenuItem having some logics for the contents. Can you try implement the component in the selector and see if it works?